components/net_traits/image_cache.rs

```diff
@@ -99,20 +99,33 @@ pub enum UsePlaceholder {
 // ImageCache public API.
 // ======================================================================
 
+pub enum ImageCacheResult {
+    Available(ImageOrMetadataAvailable),
+    LoadError,
+    Pending(PendingImageId),
+    ReadyForRequest(PendingImageId),
+}
+
 pub trait ImageCache: Sync + Send {
     fn new(webrender_api: webrender_api::RenderApi) -> Self
     where
         Self: Sized;
 
-    /// Return any available metadata or image for the given URL,
-    /// or an indication that the image is not yet available if it is in progress,
-    /// or else reserve a slot in the cache for the URL if the consumer can request images.
-    fn find_image_or_metadata(
+    /// Definitively check whether there is a cached, fully loaded image available.
+    fn get_image(&self, url: &ServoUrl, use_placeholder: UsePlaceholder) -> Option<Arc<Image>>;
+
+    /// Add a listener for the provided pending image id, eventually called by
+    /// ImageCacheStore::complete_load.
+    /// If only metadata is available, Available(ImageOrMetadataAvailable) will
+    /// be returned.
+    /// If Available(ImageOrMetadataAvailable::Image) or LoadError is the final value,
+    /// the provided listener will be dropped (consumed & not added to PendingLoad).
+    fn track_image(
         &self,
         url: ServoUrl,
         use_placeholder: UsePlaceholder,
         can_request: CanRequestImages,
```

Comment:
There are no callers using CanRequestImages::No, so we should be able to remove this argument entirely now.

Reply:
What about [here](https://github.com/servo/servo/blob/14b74609eb75cce02b14a45795379587d017d39b/components/layout/context.rs#L119)?

Reply:
Ah, I overlooked that. Pity.